sourcegraph/sourcegraph
# Add check for release branch to make sure local branch is synced with remote

- move branch check function to `ensureBranchUpToDate()` to make sure it can be done on any branch
- accomodate existing `ensureMainBranchUpToDate()`
- make sure release branch can be checked as well

Solve: https://github.com/sourcegraph/sourcegraph/issues/32783



## Test plan
@marsblkm tested locally and **manually** by doing the following:
1. Checkout to branch `3.39` and randomly commit something (to simulate the scenario)
2. Checkout to `main`
3. Merge `jm/check-local-and-remote-release-branch` into `main`.
4. Set up dry run mode and execute `yarn release release:create-candidate final`

<img width="703" alt="image" src="https://user-images.githubusercontent.com/11640534/164342889-2cf253b8-3346-4334-a951-ca97c984731f.png">
